packages/fiori/src/ViewSettingsDialog.js

```diff
@@ -667,6 +676,61 @@ class ViewSettingsDialog extends UI5Element {
 		// Invalidate
 		this._currentSettings = JSON.parse(JSON.stringify(this._currentSettings));
 	}
+
+	/**
+	 * Sets a JavaScript object, as settings to the ui5-view-settings-dialog.
+	 * This method can be used after the dialog is initially open, as the dialog need to set its initial settings.
+	 * The <code>ui5-view-settings-dialog</code> throws an event called "before-open", this can be used as trigger point.
+	 * The object should have the following format:
+	 * <code>{
+	 *	{ "sortOrder" : "Ascending", "sortBy" : "Name", "filters" : [{"Filter 1": ["Some filter 1", "Some filter 2"]}, {"Filter 2": ["Some filter 4"]}]}
+	 * }</code>
+	 * @param {string} settings A value to be set as predefined settings.
```

Comment:
**Question**: what's the reasoning behind passing `settings` param as a `String` rather than an `Object`?

I'm working on this feature ([4896](https://github.com/SAP/ui5-webcomponents/issues/4896)) for our application and passing `settings` as a `String` does not work:

```sortDialogRef?.current?.setConfirmedSettings('{ "sortOrder" : "Ascending", "sortBy" : "Name", "filters" : [{"Filter 1": ["Some filter 1", "Some filter 2"]}, {"Filter 2": ["Some filter 4"]}]}');```

However, passing it as an `Object` works.

Reply:
```
export interface ViewSettingsDialogDomRef extends ViewSettingsDialogAttributes, Ui5DomRef {
  /**
   *...
   * @param {string} settings - A value to be set as predefined settings.
   */
  setConfirmedSettings: (settings: string) => void;
  ...
}
```
Ref: https://github.com/SAP/ui5-webcomponents-react/blob/main/packages/main/src/webComponents/ViewSettingsDialog/index.tsx#L20

Reply:
There was a mistake in the documentation. You have to use it as an object. As you correctly checked, it works well with object.

Reply:
Could you please have correct person on the team update the TS strong type on `setConfirmedSettings: (settings: string) => void` : https://github.com/SAP/ui5-webcomponents-react/blob/main/packages/main/src/webComponents/ViewSettingsDialog/index.tsx#L20?